src/System.IO.FileSystem/tests/File/Move.cs

```diff
@@ -62,9 +62,20 @@ public void PathWithIllegalCharacters_Core(string invalidPath)
             testFile.Create().Dispose();
 
             if (invalidPath.Contains('\0'.ToString()))
+            {
                 Assert.Throws<ArgumentException>(() => Move(testFile.FullName, invalidPath));
+            }
             else
-                Assert.ThrowsAny<IOException>(() => Move(testFile.FullName, invalidPath));
+            {
+                if (PlatformDetection.IsInAppContainer)
+                {
+                    AssertExtensions.ThrowsAny<IOException, UnauthorizedAccessException>(() => Move(testFile.FullName, invalidPath));
```

Comment:
Curious why multiple different exceptions are possible?  It's based on the nature of the invalid path provided?  If so, should we be more fine-grained about which kind of invalid path maps to which exception type?

Reply:
F5 has blocks on traversal that you normally don't hit. I don't think we care about the fact that we get both types here, but it is nice to surface if we start seeing it out of the appcontainer.

Reply:
The test paths we use are an even mix of the two. It did not seem worth wiring through exactly which one. In some cases, you can't predict, for example attempting to create the root directory can create DirectoryNotFoundException instead of UnauthorizedAccessException; that seems like the sort of behavior that could change and not a high priority to validate. So, I left as is.